control-admin/src/main/java/fi/nls/oskari/control/admin/LayerAdminUsageCheckHandler.java

```diff
@@ -0,0 +1,72 @@
+package fi.nls.oskari.control.admin;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import fi.nls.oskari.annotation.OskariActionRoute;
+import fi.nls.oskari.control.*;
+import fi.nls.oskari.domain.map.OskariLayer;
+import fi.nls.oskari.log.LogFactory;
+import fi.nls.oskari.log.Logger;
+import fi.nls.oskari.map.layer.OskariLayerService;
+import fi.nls.oskari.service.OskariComponentManager;
+import fi.nls.oskari.service.ServiceRuntimeException;
+import fi.nls.oskari.util.ResponseHelper;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.*;
+
+@OskariActionRoute("LayerAdminUsageCheck")
+public class LayerAdminUsageCheckHandler extends RestActionHandler {
+    private static final Logger LOG = LogFactory.getLogger(LayerAdminUsageCheckHandler.class);
+
+    private static final String PARAM_LAYER_ID = "id";
+    private OskariLayerService mapLayerService;
+    private ObjectMapper mapper = new ObjectMapper();
+
+    @Override
+    public void init() {
+        try {
+            mapLayerService = OskariComponentManager.getComponentOfType(OskariLayerService.class);
+        } catch (Exception e) {
+            throw new ServiceRuntimeException("Exception occured while initializing map layer service", e);
+        }
+    }
+
+    @Override
+    public void handleGet(ActionParameters params) throws ActionException {
```

Comment:
Start this off with `params.requireAdminUser()` as first line since this is for admins only. Then we don't really need to care if we expose ids that might require some permission.